src/main/java/com/flagsmith/FlagsmithApiWrapper.java

```diff
@@ -1,176 +1,216 @@
 package com.flagsmith;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
-import java.io.IOException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.flagsmith.config.FlagsmithCacheConfig;
+import com.flagsmith.config.FlagsmithConfig;
+import com.flagsmith.flagengine.environments.EnvironmentModel;
+import com.flagsmith.flagengine.features.FeatureStateModel;
+import com.flagsmith.flagengine.identities.IdentityModel;
+import com.flagsmith.flagengine.identities.traits.TraitModel;
+import com.flagsmith.interfaces.FlagsmithCache;
+import com.flagsmith.interfaces.FlagsmithSdk;
+import com.flagsmith.models.BaseFlag;
+import com.flagsmith.threads.RequestProcessor;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
-import okhttp3.Call;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import lombok.Data;
 import okhttp3.HttpUrl;
 import okhttp3.MediaType;
 import okhttp3.Request;
 import okhttp3.RequestBody;
-import okhttp3.Response;
 
-class FlagsmithApiWrapper implements FlagsmithSdk {
+@Data
+public class FlagsmithApiWrapper implements FlagsmithSdk {
 
   private static final String AUTH_HEADER = "X-Environment-Key";
   private static final String ACCEPT_HEADER = "Accept";
+  private static final Integer TIMEOUT = 15000;
+  private static final String FLAG_KEY = "feature-flag";
+
   private final FlagsmithLogger logger;
   private final FlagsmithConfig defaultConfig;
   private final HashMap<String, String> customHeaders;
   // an api key per environment
   private final String apiKey;
+  private final RequestProcessor requestor;
+  private FlagsmithCache cache = null;
+
+  /**
+   * Instantiate with cache.
+   *
+   * @param cache cache object
+   * @param defaultConfig config object
+   * @param customHeaders custom headers list
+   * @param logger logger object
+   * @param apiKey api key
+   */
+  public FlagsmithApiWrapper(
```

Comment:
This makes me very happy that we've combined the Cached and regular ApiWrapper classes 🙌